planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp

```diff
@@ -878,180 +909,186 @@ bool NormalLaneChange::getLaneChangePaths(
 
-  const auto prepare_duration = calcPrepareDuration(current_lanes, target_lanes);
+  const auto prepare_durations = calcPrepareDuration(current_lanes, target_lanes);
 
-  for (const auto & sampled_longitudinal_acc : longitudinal_acc_sampling_values) {
-    // get path on original lanes
-    const auto prepare_velocity = std::max(
-      current_velocity + sampled_longitudinal_acc * prepare_duration,
-      minimum_lane_changing_velocity);
+  for (const auto & prepare_duration : prepare_durations) {
+    for (const auto & sampled_longitudinal_acc : longitudinal_acc_sampling_values) {
+      // get path on original lanes
+      const auto prepare_velocity = std::max(
+        current_velocity + sampled_longitudinal_acc * prepare_duration,
+        minimum_lane_changing_velocity);
 
-    // compute actual longitudinal acceleration
-    const double longitudinal_acc_on_prepare =
-      (prepare_duration < 1e-3) ? 0.0 : ((prepare_velocity - current_velocity) / prepare_duration);
+      // compute actual longitudinal acceleration
+      const double longitudinal_acc_on_prepare =
+        (prepare_duration < 1e-3) ? 0.0
+                                  : ((prepare_velocity - current_velocity) / prepare_duration);
 
-    const double prepare_length = current_velocity * prepare_duration +
-                                  0.5 * longitudinal_acc_on_prepare * std::pow(prepare_duration, 2);
+      const double prepare_length =
+        current_velocity * prepare_duration +
+        0.5 * longitudinal_acc_on_prepare * std::pow(prepare_duration, 2);
 
-    auto prepare_segment = getPrepareSegment(current_lanes, backward_path_length, prepare_length);
+      auto prepare_segment = getPrepareSegment(current_lanes, backward_path_length, prepare_length);
 
-    if (prepare_segment.points.empty()) {
-      RCLCPP_DEBUG(logger_, "prepare segment is empty!!");
-      continue;
-    }
-
-    // lane changing start getEgoPose() is at the end of prepare segment
-    const auto & lane_changing_start_pose = prepare_segment.points.back().point.pose;
-    const auto target_length_from_lane_change_start_pose = utils::getArcLengthToTargetLanelet(
-      current_lanes, target_lanes.front(), lane_changing_start_pose);
+      if (prepare_segment.points.empty()) {
+        RCLCPP_DEBUG(logger_, "prepare segment is empty!!");
+        continue;
+      }
 
-    // Check if the lane changing start point is not on the lanes next to target lanes,
-    if (target_length_from_lane_change_start_pose > 0.0) {
-      RCLCPP_DEBUG(
-        logger_, "[only new arch] lane change start getEgoPose() is behind target lanelet!!");
-      break;
-    }
+      // lane changing start getEgoPose() is at the end of prepare segment
+      const auto & lane_changing_start_pose = prepare_segment.points.back().point.pose;
+      const auto target_length_from_lane_change_start_pose = utils::getArcLengthToTargetLanelet(
+        current_lanes, target_lanes.front(), lane_changing_start_pose);
```

Comment:
[nits] Is there any guard for empty `target_lanes`?

Reply:
Empty check is [here](https://github.com/autowarefoundation/autoware.universe/blob/0b64c861f33a3b1c6dd0ab942d56e8519e4bb807/planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp#L76-L78)

But, it is more better to check empty in this function.
Is it okay to change in other PR?

Reply:
NO PROBLEM!!!